### PR TITLE
Add Public Keys API

### PR DESCRIPTION
**Pull request by obsc**
@pengwynn

Added functionality for https://developer.github.com/v3/users/keys/

Also refactored the expanding of default links into a single method in hyperlink.go. Let me know what you think of this.

### CODE REVIEW

File: octokit/hyperlink.go [hunk 1]
```diff
@@ -21,3 +21,12 @@ func (l Hyperlink) Expand(m M) (u *url.URL, err error) {
 	u, err = sawyerHyperlink.Expand(hypermedia.M(m))
 	return
 }
+
+// Expands a link with possible, otherwise it expands the default link
+func ExpandWithDefault(link *Hyperlink, defaultLink *Hyperlink, params M) (u *url.URL, err error) {
```
Comment (pengwynn) [inline octokit/hyperlink.go]: @jingweno Thoughts on this one?

File: octokit/public_keys_test.go [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+	assert.Equal(t, 8675080, key.Id)
+	assert.Equal(t, "ssh-rsa AAA...", key.Key)
+	assert.Equal(t, "https://api.github.com/user/keys/8675080", key.URL)
+	assert.Equal(t, "aKey", key.Title)
+	assert.Equal(t, true, key.Verified)
+	assert.Equal(t, &testTime, key.CreatedAt)
+}
+
+func validateNextPage(t *testing.T, result *Result) {
```
Comment (pengwynn) [inline octokit/public_keys_test.go]: This is going to collide with the one in `followers_test.go`.

**Comment by pengwynn** [comment]
@obsc One comment and then this is good to merge. :+1:

**Comment by obsc** [comment]
Fixed name collision

@pengwynn Should the rest of the codebase be refactored to use ExpandWithDefault to promote DRYer code?

**Comment by pengwynn** [comment]
> Should the rest of the codebase be refactored to use `ExpandWithDefault` to promote DRYer code?

We could explore that in a separate PR if you'd like.

**Pull request merged**